### ISSUE TITLE
Config file of instance segmentation for linear probing on Scannet with Sonata

### ISSUE THREAD

**Issue by binyang97** [Pointcept/Pointcept#499]
Hi Xiaoyang,

Thank you for releasing the pre-trained weights of SONATA. I tried using them for linear probing on ScanNet instance segmentation, but my results differ from those reported in the paper.

My evaluation gave the following numbers:

mAP / AP50 / AP25: 27.42 / 47.10 / 65.60

Whereas the paper reports:

30.7 / 53.9 / 72.6

Could you kindly let me know if there are specific settings or implementation details I might be missing? Currently, I simply adapted the config file of instance segmentation with PTv3: [https://github.com/Pointcept/Pointcept.git/configs/scannet/insseg-pointgroup-v1m2-0-ptv3-base.py](url)

Thank you!

Best, 
Bin Yang

**Comment by Gofinge** [comment]
Hey, could you share your modified config for reference?

**Comment by binyang97** [comment]
Sure, here is my config file:


_base_ = ["../_base_/default_runtime.py"]


batch_size = 24  # bs: total bs in all gpus
num_worker = 8
mix_prob = 0
clip_grad = 3.0
empty_cache = False
enable_amp = True
enable_wandb = False


class_names = [
    "wall",
    "floor",
    "cabinet",
    "bed",
    "chair",
    "sofa",
    "table",
    "door",
    "window",
    "bookshelf",
    "picture",
    "counter",
    "desk",
    "curtain",
    "refridgerator",
    "shower curtain",
    "toilet",
    "sink",
    "bathtub",
    "otherfurniture",
]
num_classes = 20
segment_ignore_index = (-1, 0, 1)


model = dict(
    type="PG-v1m2",
    backbone=dict(
        type="PT-v3m2",
        in_channels=9,
        order=("z", "z-trans", "hilbert", "hilbert-trans"),
        stride=(2, 2, 2, 2),
        enc_depths=(3, 3, 3, 12, 3),
        enc_channels=(48, 96, 192, 384, 512),
        enc_num_head=(3, 6, 12, 24, 32),
        enc_patch_size=(1024, 1024, 1024, 1024, 1024),
        mlp_ratio=4,
        qkv_bias=True,
        qk_scale=None,
        attn_drop=0.0,
        proj_drop=0.0,
        drop_path=0.3,
        shuffle_orders=True,
        pre_norm=True,
        enable_rpe=False,
        enable_flash=True,
        upcast_attention=False,
        upcast_softmax=False,
        traceable=False,
        mask_token=False,
        enc_mode=True,
        freeze_encoder=False,
    ),
    # backbone_out_channels=992,
    backbone_out_channels=1232,
    semantic_num_classes=num_classes,
    semantic_ignore_index=-1,
    segment_ignore_index=segment_ignore_index,
    instance_ignore_index=-1,
    cluster_thresh=1.5,
    cluster_closed_points=300,
    cluster_propose_points=100,
    cluster_min_points=50,
    criteria=[
        dict(type="CrossEntropyLoss", loss_weight=1.0, ignore_index=-1),
        dict(type="LovaszLoss", mode="multiclass", loss_weight=1.0, ignore_index=-1),
    ],
    freeze_backbone=True,
)


epoch = 800 
optimizer = dict(type="AdamW", lr=0.002, weight_decay=0.02)
scheduler = dict(
    type="OneCycleLR",
    max_lr=[0.002, 0.0002],
    pct_start=0.05,
    anneal_strategy="cos",
    div_factor=10.0,
    final_div_factor=1000.0,
)
param_dicts = [dict(keyword="block", lr=0.0002)]


dataset_type = "ScanNetDataset"
data_root = "/home/yai5fe/project_folder/scannet"

data = dict(
    num_classes=20,
    ignore_index=-1,
    names=class_names,
    train=dict(
        type=dataset_type,
        split="train",
        data_root=data_root,
        transform=[
            dict(type="CenterShift", apply_z=True),
            dict(
                type="RandomDropout", dropout_ratio=0.2, dropout_application_ratio=1.0
            ),
            # dict(type="RandomRotateTargetAngle", angle=(1/2, 1, 3/2), center=[0, 0, 0], axis="z", p=0.75),
            dict(type="RandomRotate", angle=[-1, 1], axis="z", center=[0, 0, 0], p=0.5),
            dict(type="RandomRotate", angle=[-1 / 64, 1 / 64], axis="x", p=0.5),
            dict(type="RandomRotate", angle=[-1 / 64, 1 / 64], axis="y", p=0.5),
            dict(type="RandomScale", scale=[0.9, 1.1]),
            # dict(type="RandomShift", shift=[0.2, 0.2, 0.2]),
            dict(type="RandomFlip", p=0.5),
            dict(type="RandomJitter", sigma=0.005, clip=0.02),
            dict(type="ElasticDistortion", distortion_params=[[0.2, 0.4], [0.8, 1.6]]),
            dict(type="ChromaticAutoContrast", p=0.2, blend_factor=None),
            dict(type="ChromaticTranslation", p=0.95, ratio=0.05),
            dict(type="ChromaticJitter", p=0.95, std=0.05),
            # dict(type="HueSaturationTranslation", hue_max=0.2, saturation_max=0.2),
            # dict(type="RandomColorDrop", p=0.2, color_augment=0.0),
            dict(
                type="GridSample",
                grid_size=0.02,
                hash_type="fnv",
                mode="train",
                return_grid_coord=True,
                keys=("coord", "color", "normal", "segment", "instance"),
            ),
            dict(type="SphereCrop", point_max=102400, mode="random"),
            dict(type="CenterShift", apply_z=False),
            dict(type="NormalizeColor"),
            dict(
                type="InstanceParser",
                segment_ignore_index=segment_ignore_index,
                instance_ignore_index=-1,
            ),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment", "instance", "instance_centroid", "bbox"),
                feat_keys=("coord", "color", "normal"),
            ),
        ],
        test_mode=False,
    ),
    val=dict(
        type=dataset_type,
        split="val",
        data_root=data_root,
        transform=[
            dict(type="CenterShift", apply_z=True),
            dict(
                type="Copy",
                keys_dict={
                    "coord": "origin_coord",
                    "segment": "origin_segment",
                    "instance": "origin_instance",
                },
            ),
            dict(
                type="GridSample",
                grid_size=0.02,
                hash_type="fnv",
                mode="train",
                return_grid_coord=True,
                return_inverse=True,
                keys=("coord", "color", "normal", "segment", "instance"),
            ),
            dict(type="CenterShift", apply_z=False),
            dict(type="NormalizeColor"),
            dict(
                type="InstanceParser",
                segment_ignore_index=segment_ignore_index,
                instance_ignore_index=-1,
            ),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment", "origin_segment", "inverse", "instance",
                    "origin_coord",
                    "origin_instance",
                    "instance_centroid",
                    "bbox",),
                feat_keys=("coord", "color", "normal"),
                offset_keys_dict=dict(offset="coord", origin_offset="origin_coord"),
            ),
        ],
        test_mode=False,
    ),
    test=dict(
        type=dataset_type,
        split="val",
        data_root=data_root,
        transform=[
            dict(type="CenterShift", apply_z=True),
            dict(
                type="Copy",
                keys_dict={
                    "coord": "origin_coord",
                    "segment": "origin_segment",
                    "instance": "origin_instance",
                },
            ),
            dict(
                type="GridSample",
                grid_size=0.02,
                hash_type="fnv",
                mode="train",
                return_grid_coord=True,
                keys=("coord", "color", "normal", "segment", "instance"),
            ),
            dict(type="CenterShift", apply_z=False),
            dict(type="NormalizeColor"),
            dict(
                type="InstanceParser",
                segment_ignore_index=segment_ignore_index,
                instance_ignore_index=-1,
            ),
            dict(type="ToTensor"),
            dict(
                    type="Collect",
                    keys=(
                        "coord",
                        "grid_coord",
                        "segment",
                        "instance",
                        "origin_coord",
                        "origin_segment",
                        "origin_instance",
                        "instance_centroid",
                        "bbox",
                        "name",
                    ),
                    feat_keys=("coord", "color", "normal"),
                    offset_keys_dict=dict(offset="coord", origin_offset="origin_coord"),
                ),
        ],
        test_mode=False, # TODO: design test mode for ins seg, e.g. TTA
    ),
)



hooks = [
    dict(
        type="CheckpointLoader",
        keywords="module",
        replacement="module.backbone",
    ),
    dict(type="IterationTimer", warmup_iter=2),
    dict(type="InformationWriter", interval=200),
    dict(
        type="InsSegEvaluator",
        segment_ignore_index=segment_ignore_index,
        instance_ignore_index=-1,
        interval=100,
    ),
    dict(type="CheckpointSaver", save_freq=None),
    dict(type="PreciseEvaluator", test_last=False),
]


test = dict(
    type="InsSegTester",
    segment_ignore_index=segment_ignore_index,
    instance_ignore_index=-1,
    verbose=False,
)

**Comment by Gofinge** [comment]
Oh, yes. I remember: `cluster_closed_points=300` -> `cluster_closed_points=600`, `cluster_propose_points=100` -> `cluster_propose_points=200`, I kept `batch_size=12`, also I added support to `mix_prob=0.8` (maybe I should add these changes to a PR). This setting should be helpful for benchmarking with more categories (e.g., ScanNet200). I didn't spend too much time in each benchmark for the ins seg, just parallel run some setups.

**Comment by binyang97** [comment]
Thank you so much! Now I get some results much closer to those in the paper. I will retrain with `mix_prob=0.8` and see how it is going. Btw, do you reduce the batch size for semantic segmentation as well?

**Comment by Gofinge** [comment]
I think no much difference after well pre-training

**Comment by binyang97** [comment]
Thank you for the quick reply. I'm able to reproduce the results now.

**Comment by LandDreamer** [comment]
> Thank you so much! Now I get some results much closer to those in the paper. I will retrain with `mix_prob=0.8` and see how it is going. Btw, do you reduce the batch size for semantic segmentation as well?

Hi! I’m also interested in reproducing the results. Could you please share the revisions you made? Thank you!
